helpers/framed.js

```diff
@@ -0,0 +1,31 @@
+let framed;
+
+export async function isFramed() {
+	if (framed !== undefined) return framed;
+
+	if ((window.D2L && window.D2L.IsNotAnIframedApp) || window === window.parent) {
+		framed = false;
+		return framed;
+	}
+
+	const handleIsFramedResponse = (evt, resolve) => {
+		if (!evt || !evt.data || !evt.data.isFramed) return;
+		window.removeEventListener('message', evt => handleIsFramedResponse(evt, resolve), false);
```

Comment:
Double-check that this is actually getting removed. I've only ever seen this done by passing a single argument (e.g. `removeEventListener('eventName', callbackName, false)`) -- never an arrow function. I have a feeling this is creating a new arrow function that isn't going to match the one that was created when it was registered.

Reply:
Yeah, I think you're right. This will end up defining a new function.